File: src/php/db-get.php
<?php
declare(strict_types = 1);

$db = new mysqli('p:localhost', 'loadtesting', 'loadtesting', 'loadtesting', 3306, '/var/run/mysqld/mysqld.sock');
if ($db->connect_errno) {
    http_response_code(500);
    die();
}

$dbq = $db->query('SELECT HEX(hexId), incrementValue, textField FROM `table1`');
if (!$dbq) {
    http_response_code(500);
    die();
}

$data = $dbq->fetch_all(MYSQLI_ASSOC);

echo json_encode($data);

File: src/resty/db-get.lua
local mysql = require "resty.mysql"
local cjson = require "cjson"

local db = mysql:new()

local ok = db:connect({
    path = '/var/run/mysqld/mysqld.sock',
    database = 'loadtesting',
    user = 'loadtesting',
    password = 'loadtesting'
})
if not ok then
    ngx.exit(500)
end

local dbd = db:query('SELECT HEX(hexId), incrementValue, textField FROM `table1`')
if not dbd then
    ngx.exit(500)
end

local ok = db:set_keepalive(5000, 1000)
if not ok then
    ngx.exit(500)
end

ngx.say(cjson.encode(dbd))
